examples/arduino/LVGL_Arduino/LVGL_Arduino.ino

```diff
@@ -89,6 +104,8 @@ void setup()
 #if LV_USE_TFT_ESPI
     /*TFT_eSPI can be enabled lv_conf.h to initialize the display in a simple way*/
     disp = lv_tft_espi_create(TFT_HOR_RES, TFT_VER_RES, draw_buf, sizeof(draw_buf));
+    ((lv_tft_espi_t *)(lv_display_get_driver_data(disp)))->tft->setRotation(TFT_ROTATION);
```

Comment:
Did this work for you when you tested with different values of `TFT_ROTATION`? Changing the rotation would swap the hor+ver resolution, wouldn't it? I think it would be better to rotate the display using `lv_display_set_rotation` so the user doesn't have to manually swap the hor+ver resolution when they change the rotation.

```suggestion
    /*TFT_eSPI can be enabled lv_conf.h to initialize the display in a simple way*/
    disp = lv_tft_espi_create(TFT_HOR_RES, TFT_VER_RES, draw_buf, sizeof(draw_buf));
    lv_display_set_rotation(disp, TFT_ROTATION);
```

Also update `#define TFT_ROTATION 0` to `#define TFT_ROTATION LV_DISPLAY_ROTATION_0`

Reply:
Well, I do not fully understand what `lv_display_set_rotation()` is really doing. I none of my tests it actually rotated what I can see on the TFT.
The only way I could get rotated content on the tft, is by setting rotation in `TFT_eSPI`. But the rotation of `TFT_eSPI` is hardcoded in "lv_tft_espi.cpp" to `dsc->tft->setRotation(3)`, which does not work in my case. So I looked for a way to set it.
Maybe we need a way to provide the needed rotation to function `lv_tft_espi_create()` as a parameter.

For my display to work, I have to set
```
#define TFT_HOR_RES   240
#define TFT_VER_RES   320

 ((lv_tft_espi_t *)(lv_display_get_driver_data(disp)))->tft->setRotation(0);
```

Reply:
Aha, based on @kisvegabor's review, I see that `lv_display_set_rotation` is not the whole solution. It seems like adding the event callback he described would do it. So `tft->setRotation` would be invoked, but inside the driver lv_tft_espi.cpp as a result of `lv_display_set_rotation` being called by the user.

Are you up to the task? This is a starting point you can begin testing from.
```diff
diff --git a/src/drivers/display/tft_espi/lv_tft_espi.cpp b/src/drivers/display/tft_espi/lv_tft_espi.cpp
index 64e3ee6ab..adf1ef641 100644
--- a/src/drivers/display/tft_espi/lv_tft_espi.cpp
+++ b/src/drivers/display/tft_espi/lv_tft_espi.cpp
@@ -26,6 +26,7 @@ typedef struct {
  *  STATIC PROTOTYPES
  **********************/
 static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map);
+static void resolution_changed_event_cb(lv_event_t * e);
 
 /**********************
  *  STATIC VARIABLES
@@ -53,9 +54,10 @@ lv_display_t * lv_tft_espi_create(uint32_t hor_res, uint32_t ver_res, void * buf
 
     dsc->tft = new TFT_eSPI(hor_res, ver_res);
     dsc->tft->begin();          /* TFT init */
-    dsc->tft->setRotation(3);   /* Landscape orientation, flipped */
+    dsc->tft->setRotation(0);
     lv_display_set_driver_data(disp, (void *)dsc);
     lv_display_set_flush_cb(disp, flush_cb);
+    lv_display_add_event_cb(disp, resolution_changed_event_cb, LV_EVENT_RESOLUTION_CHANGED, NULL);
     lv_display_set_buffers(disp, (void *)buf, NULL, buf_size_bytes, LV_DISPLAY_RENDER_MODE_PARTIAL);
     return disp;
 }
@@ -80,4 +82,33 @@ static void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_m
 
 }
 
+static void resolution_changed_event_cb(lv_event_t * e)
+{
+    lv_display_t * disp = (lv_display_t *)lv_event_get_target(e);
+    lv_tft_espi_t * dsc = (lv_tft_espi_t *)lv_display_get_driver_data(disp);
+    int32_t hor_res = lv_display_get_horizontal_resolution(disp);
+    int32_t ver_res = lv_display_get_vertical_resolution(disp);
+    lv_display_rotation_t rot = lv_display_get_rotation(disp);
+
+    /* TODO: implement resolution change */
+    LV_UNUSED(hor_res);
+    LV_UNUSED(ver_res);
+
+    /* handle rotation */
+    switch(rot) {
+        case LV_DISPLAY_ROTATION_0:
+            dsc->tft->setRotation(0);   /* Portrait orientation */
+            break;
+        case LV_DISPLAY_ROTATION_90:
+            dsc->tft->setRotation(1);   /* Landscape orientation */
+            break;
+        case LV_DISPLAY_ROTATION_180:
+            dsc->tft->setRotation(2);   /* Portrait orientation, flipped */
+            break;
+        case LV_DISPLAY_ROTATION_270:
+            dsc->tft->setRotation(3);   /* Landscape orientation, flipped */
+            break;
+    }
+}
+
 #endif /*LV_USE_TFT_ESPI*/
```

Reply:
Ah, ok, now I understand, thanks.
I can confirm that this event handler is working great. With that we can set the rotation.

The TODO for the resolution change is something that cannot be achieved with TFT_eSPI, I think. There are no setters for the resolution, you can only set it in the constructor.

Ok, I think you found a much more elegant solution than I did.
Is there anything you want me to provide? A changed PR? Or do you want to do it on your own? @liamHowatt you already did all the work :-)

Reply:
It's great that there is an agreement :slightly_smiling_face: 

I think it's better if @KlausMu adds the discussed changes as he has an envirement to test it too.

Reply:
Ok, I added the changes and comitted them. Please be aware that [this PR](https://github.com/lvgl/lvgl/pull/6001) also needs to be merged, otherwise the example will not even compile (the original example never compiled).

Reply:
Assuming you mean this one #5999, it has been merged. Thanks for the heads-up. If you want you can [rebase or merge](https://www.atlassian.com/git/tutorials/merging-vs-rebasing) if you feel like it. Just for your convenience -- totally optional in this case.